x-pack/plugins/infra/public/hooks/use_http_request.tsx

```diff
@@ -19,18 +20,20 @@ export function useHTTPRequest<Response>(
   body?: string | null,
   decode: (response: any) => Response = (response) => response,
   fetch?: HttpHandler,
-  toastDanger?: (input: ToastInput) => void
+  toastDanger?: (input: ToastInput) => void,
+  abortable = false
```

Comment:
**thought:** this is out of the scope of this PR, but I see the argument list for the useHTTPRequest function is going a bit out of control...
It would be an excellent opportunity to refactor it replacing [introducing a parameter object](https://refactoring.com/catalog/introduceParameterObject.html) to make it more readable and easier to consume it.
As far as I can see, we use it a reasonable amount of times, so the refactor should be a quick one.